site/content/entry/yuzu-progress-report-may-2021/index.md

```diff
@@ -0,0 +1,349 @@
+as they are the programs that run whenever a game needs the kernel to perform operations that require elevated permissions.
+
+With the introduction of the firmware version `12.0.0`, the protocol of [Inter-Process Communication](https://en.wikipedia.org/wiki/Inter-process_communication) (`IPC`) 
+has also been updated.
+bunnei worked on [various improvements to the IPC and session management](https://github.com/yuzu-emu/yuzu/pull/6299) with the aim to support `TIPC` — the new protocol.
+
+`IPC` is a mechanism provided by the OS and used by processes to communicate between themselves, usually to manage or cooperate in the processing of some shared resource — such 
+as a memory region, a file, etc.
+A `session` in this context refers to an object that is created to manage the information exchange between a client (i.e. games) and a server (services that the game uses to 
+render graphics, play audio, get user input, etc.).
+With these changes, the code in charge of `session management` has been updated to support the new `IPC` protocol, additionally fixing a number of inaccuracies in some of the 
+error-checking functions.
+bunnei has also greatly simplified the original `IPC` code, which should improve the memory usage and performance of these operations.
+
+As a consequence of the changes introduced in the previous `KAutoObject` PR, the maximum values for many of the kernel objects have started to be enforced. This resulted in 
+crashes in games that open and close sessions often, such as `Nintendo Labo`, `Pokémon Sword`, and `Pokémon Shield`, due to yuzu not managing sessions correctly.
+The root problem was also the cause of small memory leaks that hadn't been noticed until now, since the sessions weren't being properly closed and remained in memory.
+For this reason, bunnei implemented mechanisms for moving and closing objects in a [follow-up PR](https://github.com/yuzu-emu/yuzu/pull/6347), which corrected the way the 
+session counter and its opening and closing operations worked.
+
+There are still many other kernel objects to migrate and, for this reason, bunnei introduced a PR to make the 
+[KSlabHeap use both guest and host allocations](https://github.com/yuzu-emu/yuzu/pull/6373), 
+as this will facilitate the process while missing functions and other structures are being implemented.
+
+Simply put, the [slab heap](https://en.wikipedia.org/wiki/Slab_allocation) is a structure used to store kernel elements more efficiently in memory on a [linked list](https://en.wikipedia.org/wiki/Linked_list), based on their size — all the slots in a slab of memory have the same size, and there can be lists of different sizes to store different objects, too.
+Instead of allocating and deallocating memory, the kernel marks the nodes on the list as "used" or "free", so when it needs to store a new object, it looks for a free slot and overwrites the data there.
```

Comment:
```suggestion
Instead of allocating and deallocating memory, the kernel marks the nodes on the list as "used" or "free," so when it needs to store a new object, it looks for a free slot and overwrites the data there.
```